apps/studio/src/components/PageEditor/LinkEditorModal.tsx

```diff
@@ -206,7 +199,13 @@ interface LinkEditorModalProps {
   onSave: (linkText: string, linkHref: string) => void
   isOpen: boolean
   onClose: () => void
-  linkTypes: LinkTypeMapping
+  linkTypes: Record<
+    string,
+    {
+      icon: IconType
+      label: Capitalize<LinkTypes>
+    }
+  >
```

Comment:
use `LinkTypeMapping` instead?

Reply:
i actually tried this and it didn't _quite_ work. this is because when we say `Record<LinkTypes, string>`, actually what we mean is a 1:1 mapping (ie, every type in `LinkType` **must** have a corresponding string) which isn't quite what we want here. what we want is "some subset of LinkTypes to string" - i considered using `Partial<Record<LinkTypes, string>>`, but unfortunately that means that **every type** is now optional. hence, i just went iwth string for now